src/main/java/org/example/Main.java

```diff
@@ -17,83 +13,117 @@
-                    medication.setDosage_form(dosage_form);
-                    medications.add(medication);
-
-                    statement.setString(1, brandName);
-                    statement.setString(2, strength);
-                    statement.setString(3, dosage_form);
-                    statement.execute();
-//                    for (JsonElement brandNameElement : brandName) {
-//                       // String brandName = brandNameElement.getAsString();
-//
-//                        Medication medication = new Medication();
-//                      //  medication.setBrand_name(brandName);
-//                        medications.add(medication);
-//
-//                       // statement.setString(1, brandName);
-//                       // statement.execute();
-//                    }
-                }
-                for (String brand : brandNameList) {
-                    System.out.println(brand);
-                }
-                for(Medication medication : medications){
-                    System.out.println(medication.toString());
-                }
-
-           statement.close();
-            jDbCconnection.close();
-        } else {
-            System.out.println("Error retrieving medication data. Response code: " + responseCode);
+  public static void main(String[] args) throws Exception {
+
+    String apiUrl = "https://api.fda.gov/drug/drugsfda.json?&limit=1000";
+
+    URL url = new URL(apiUrl);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+
+    int responseCode = connection.getResponseCode();
+    if (responseCode == HttpURLConnection.HTTP_OK) {
+      BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+      String inputLine;
+      StringBuffer response = new StringBuffer();
+      while ((inputLine = in.readLine()) != null) {
+        response.append(inputLine);
+      }
+      in.close();
+
+      JsonObject jsonObject = new Gson().fromJson(response.toString(), JsonObject.class);
+      JsonArray results = jsonObject.getAsJsonArray("results");
+
+      Class.forName("org.postgresql.Driver");
+
+      Connection jDbCconnection =
+          DriverManager.getConnection("jdbc:postgresql://localhost:5432/dex");
+      PreparedStatement statement =
+          jDbCconnection.prepareStatement(
+              "INSERT INTO medication_with_uuid (brand_name, strength, dosage_form) VALUES (?,?,?)");
+
+      List<String> brandNameList = new ArrayList<>();
+      List<Medication> medications = new ArrayList<>();
+      for (JsonElement resultElement : results) {
+        JsonObject result = resultElement.getAsJsonObject();
+        // products may be null
+        Optional<JsonArray> maybeProducts = Optional.ofNullable(result.getAsJsonArray("products"));
+        if (maybeProducts.isEmpty()) {
+          continue;
+        }
+        // if we're here that means products are good to go
```

Comment:
lol openfda sucks so bad that "products" may be null. (whats the point then?). Anyways, im checking here that there's even a products array to begin with.. if not we'll just continue.

Reply:
Haha really? That is annoying. Are there a lot of null checks?

Reply:
more than i'd like to deal with.. lol.